another random maint expansion: making maint a tad more dangerous

zeskorion opened the pull request:
## About The Pull Request
- adds 34 new random maint rooms, credit to GizkaFreechman for most. 
- new gimmick job, the VIP. the VIP is rich.
- adds random maint traps: invisible traps that may or may not spawn. they range from breaking all the lights in a room to spawning mobs nearby
- also removes knockdown from faithless while i was in his files cuz i forgot it awhile ago

## Why It's Good For The Game
I hate you all, and i want to remind you that no one is fucking safe

## Changelog
:cl: Zeskorion, GizkaFreechman, thatguythere03
add: 28 random maint rooms by GizkaFreechman, edited by myself
add: Random maint traps
add: cluwne rune, which can be used by cultists in a manner similar to an invocation rune, if the Honkmother has blessed one of the invokers with her presence
add: new random maint job, the VIP. he's rich, apparently important, and actually probably quite useless.
imageadd: sprites for new job and cluwne rune by thatguythere03
tweak: edited most random maint maps to make maint easier to get through (changing table placements and such)
balance: faithless no longer has a 12% chance to stun for 6 seconds on hit
/:cl:

ike709 commented on code/game/objects/effects/mainttraps.dm:
```diff
@@ -0,0 +1,356 @@
+		inuse = TRUE
+	var/list/possibletraps = list()
+	for(var/obj/effect/trap/nexus/payload in oview(10, src))
+		possibletraps += payload
```
This list should probably just be populated once in LateInitialize rather than every time. `oview()` is expensive.

ike709 commented:
Christ I had to stop looking at the code. You use `view()` and `oview()` *way* too much.

ike709 commented:
You should look at `fields.dm` and consider using fields for this. They shouldn't be too expensive since they shouldn't really be moving or processing.

zeskorion commented:
Aight, ill see about that when i get home them

zeskorion commented:
Dammit gizka stop using step vars

imw-Oculus commented:
> also removes knockdown from faithless while i was in his files cuz i forgot it awhile ago
> new gimmick job, the VIP. the VIP is rich.

What do these have to do with Maint? Atomize your PR's. This is the third PR of yours in a row where you've just blatantly threw in completely unrelated things into a PR. Your PR's aren't going to get merged if they're not atomized.

> adds 34 new random maint rooms, credit to GizkaFreechman for most.

This is great! In concept, at least - haven't reviewed the particular rooms yet.

> adds random maint traps: invisible traps that may or may not spawn. they range from breaking all the lights in a room to spawning mobs nearby

I like the spirit of adding traps - but I don't think they should be invisible. Give the player an opportunity to avoid them, otherwise it's just "Whoops, looks like you made the mistake of trying to explore maint! Time to die.". Good systems give the player an opportunity to interact with them.

zeskorion commented:
> > also removes knockdown from faithless while i was in his files cuz i forgot it awhile ago
> > new gimmick job, the VIP. the VIP is rich.
> 
> What do these have to do with Maint? Atomize your PR's. This is the third PR of yours in a row where you've just blatantly threw in completely unrelated things into a PR. Your PR's aren't going to get merged if they're not atomized.

nah i saw it while doing spawnercode, i'd missed it in a previous pr. Gimmick jobs, however, are a random maint thing. every random maint pr since they were introduced has had one

> > adds random maint traps: invisible traps that may or may not spawn. they range from breaking all the lights in a room to spawning mobs nearby
> 
> I like the spirit of adding traps - but I don't think they should be invisible. Give the player an opportunity to avoid them, otherwise it's just "Whoops, looks like you made the mistake of trying to explore maint! Time to die.". Good systems give the player an opportunity to interact with them.

none of the traps will directly kill the players. There are, at the moment, four kinds of trap: lightbreaker, which breaks lights, doorbolter, which bolts doors for about ten seconds, spawners, which try to spawn a mob out of sight after being triggered a certain amount of times, and cluwne curse, which is quite rare, and spawns a nonlethal floorcluwne

utilbird commented:
>non lethal floor clown

Lame

zeskorion commented:
> > non lethal floor clown
> 
> Lame

you can make it lethal through a very difficult and convoluted ritual

TeomanTheGreat commented:
> 
> 
> > > non lethal floor clown
> > 
> > 
> > Lame
> 
> you can make it lethal through a very difficult and convoluted ritual

convoluted ritual you say? explain

zeskorion commented:
> > > > non lethal floor clown
> > > 
> > > 
> > > Lame
> > 
> > 
> > you can make it lethal through a very difficult and convoluted ritual
> 
> convoluted ritual you say? explain

first, you need to find one of the incredibly rare cluwne runes. ill let you figure out the rest

utilbird commented:
Could give VIP some very expensive item that cargo will probably want, with some mood debuff and/or crew objective red text if they lose it

zeskorion commented:
> Could give VIP some very expensive item that cargo will probably want, with some mood debuff and/or crew objective red text if they lose it

Their pda is worth 500, and both envirosuit and helmet for vip plasmaman are worth 4500 each

PowerfulBacon commented on code/modules/mob/living/simple_animal/hostile/faithless.dm:
```diff
@@ -34,11 +34,3 @@
 
 	do_footstep = TRUE
 	hardattacks = TRUE
-
-/mob/living/simple_animal/hostile/faithless/AttackingTarget()
-	. = ..()
-	if(. && prob(12) && iscarbon(target))
-		var/mob/living/carbon/C = target
-		C.Paralyze(60)
-		C.visible_message("<span class='danger'>\The [src] knocks down \the [C]!</span>", \
-				"<span class='userdanger'>\The [src] knocks you down!</span>")
```
Why is this removed?

zeskorion replied:
I was in the code for it and missed it in a previous pr so i decided to hit it

PowerfulBacon commented:
Is this ready for a testmerge?

zeskorion commented:
> Is this ready for a testmerge?

Yes

PowerfulBacon commented on code/game/objects/effects/mainttraps.dm:
```diff
@@ -0,0 +1,371 @@
+		return FALSE
+	else 
+		inuse = TRUE
+	if(!possibletraps.len)
```
!LAZYLEN(possibletraps)

PowerfulBacon commented on code/game/objects/effects/mainttraps.dm:
```diff
@@ -0,0 +1,371 @@
+	var/grounded = FALSE //does it ignore fliers
+	var/pick_style = PICK_STYLE_ORDERED
+	var/requirehuman = TRUE
+	var/list/possibletraps = list()
```
Apparently it's faster to initialize lists in the Initialize proc of objects than in the definition

PowerfulBacon commented:
Testmerged onto sage

zeskorion committed:
adds the basic for maint traps

adds more trap shit, adds cluwne rune, removes faithless knockdown

new maint rooms by myself and gizka, new gimmick job, etc

hides traps betterer

fixes laggy code, stepvars

no more cult clowns, changes some weights

some balance shit

small update

qwertyquerty commented:
Fix conflicts

zeskorion committed:
readds icons

PowerfulBacon commented:
Too many maps so mapdiffs isn't working, but from watching it all of the afters generated fine and it worked when testmerged, so going to assume this to be ok
